### ISSUE TITLE
Is it possible to remove the limit on  MI_ALIGNMENT_MAX?

### ISSUE THREAD

**Issue by Undxxx** [microsoft/mimalloc#628]
I'm having some problems with the restriction on max alignment size MI_ALIGNMENT_MAX in mimalloc. I know that this restriction is to reduce memory waste, but in our project we use memalign with alignment set to 4MiB and size set to 4MiB, which helps we get our object header fast by masking bits, just like what mimalloc does with segments. I have read this issue #576 and I know that dev-slice branch is expanding this limit to 16MiB. Still I'm wondering could we just remove this limit, and do any alignment allocation regardless of however big the alignment is? This might be a potential requirement in our project. If it is possible, what change should I do to mimalloc code?

**Comment by daanx** [comment]
I agree, it would be nice to at least allow it; I think it can be done but I need to ponder it a bit more. One work around would be to have a special aligned allocation call for large alignments  (using the OS directly) with a special free function (calling the OS free) but I would like to avoid that for now.

**Comment by DaPhreak** [comment]
I ran into the same problem when using the std::pmr pooled memory resources with mimalloc.

Yes, i can increase the MI_ALIGNMENT_MAX to some value myself as a workaround.
Or just remove all memory pools we use in our source (what i just did).

But i still cannot be sure what alignment all other libraries we use need as maximum (and we use loads...).
If they use e.g. the std::pmr pools by default, it can soon be much higher than MI_ALIGNMENT_MAX.

**Comment by daanx** [comment]
The latest `dev` and `dev-slice` branches remove the aligned allocation limit and should work with the `std::pmr` pools. Let me know if it works for you :-)
